### PR TITLE
Fix two VAAPI crash bugs

### PR DESCRIPTION
**Pull request by MaxKellermann**

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp [hunk 1]
```diff
@@ -1873,8 +1873,12 @@ void COutput::Flush()
   {
     if (msg->signal == COutputDataProtocol::NEWFRAME)
     {
-      CVaapiDecodedPicture pic = *reinterpret_cast<CVaapiDecodedPicture*>(msg->data);
-      m_config.videoSurfaces->ClearRender(pic.videoSurface);
+      CPayloadWrap<CVaapiDecodedPicture> *payload;
+      payload = dynamic_cast<CPayloadWrap<CVaapiDecodedPicture>*>(msg->payloadObj.get());
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: declaration and assignment can be done in the same line

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp [hunk 2]
```diff
@@ -1873,8 +1873,12 @@ void COutput::Flush()
...
+      if (payload) {
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: curly bracket has to go into separate line

Comment (MaxKellermann) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: Right, missed that one. Fixed.

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp (hunk 1 above)
Comment (MaxKellermann) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: I know, I'd always do it that way (and use `auto`), but this is copy'n'pasted from 61dfb26640761e217cd192529fa65fae2117aa2c so I figured this was the common Kodi coding style. Do you want me to change it?

Comment (Rechi) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: Just keep it like you have done it now.
There is no strict rule for deceleration and assignment, but in usually it is done in one line.

**Comment by pkerling** [comment]
Can't really say much here, this is more @FernetMenta's territory. More information on the crash in e87b9ced (debug log etc.) would certainly be nice. I assume that if `m_decodedPics` runs empty, something in the state machine has gone wrong already. I remember @FernetMenta saying that I should not check for `empty()` specifically for that reason.

**Comment by MaxKellermann** [comment]
Some context: both crashes occured following a VDR crash, i.e. the VNSI connection has been reset and VNSI playback is being aborted. This happens to me all the time - every time VDR crashes (happens often!), Kodi crashes as well (and more Kodi crash bugs related to that remain to be analyzed and fixed).

**Comment by FernetMenta** [comment]
Please drop the commit I commented on. Then we can merge the other

**Review by FernetMenta — CHANGES_REQUESTED** [review]

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp [hunk 3]
```diff
@@ -2530,6 +2535,11 @@ bool CVppPostproc::AddPicture(CVaapiDecodedPicture &pic)
 
 bool CVppPostproc::Filter(CVaapiProcessedPicture &outPic)
 {
+  if (m_decodedPics.empty())
```
[the remark this answers is not in the document]
Comment (MaxKellermann) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp]: Dropped.

**Review by FernetMenta — APPROVED** [review]

**Pull request merged**